src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckDiagnosticAnalyzer.cs

```diff
@@ -284,7 +284,7 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
         {
             localSymbol = semanticModel.GetSymbolInfo(identifier).Symbol as ILocalSymbol;
             declarator = localSymbol?.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax() as VariableDeclaratorSyntax;
-            return declarator != null;
+            return localSymbol != null && declarator != null;
```

Comment:
Why was this needed? Seems `declarator` cannot be non-null unless `localSymbol` is also non-nulll.

Reply:
hrmm..  compiler doesn't seem smart enough here and thinks its posisble that localSymbol might not be assigned on out  here.